internal/benchrunner/runners/system/runner.go

```diff
@@ -0,0 +1,768 @@
+	if err != nil {
+		return nil, err
+	}
+
+	packagePolicy, err := r.createPackagePolicy(pkgManifest, policy)
+	if err != nil {
+		return nil, err
+	}
+
+	r.deletePolicyHandler = func() error {
+		var merr multierror.Error
+
+		logger.Debug("deleting benchmark package policy...")
+		if err := r.options.KibanaClient.DeletePackagePolicy(*packagePolicy); err != nil {
+			merr = append(merr, fmt.Errorf("error cleaning up benchmark package policy: %w", err))
+		}
+
+		logger.Debug("deleting benchmark policy...")
+		if err := r.options.KibanaClient.DeletePolicy(*policy); err != nil {
+			merr = append(merr, fmt.Errorf("error cleaning up benchmark policy: %w", err))
+		}
+
+		if len(merr) > 0 {
+			return merr
+		}
+
+		return nil
+	}
+
+	return policy, nil
+}
+
+func (r *runner) createPackagePolicy(pkgManifest *packages.PackageManifest, p *kibana.Policy) (*kibana.PackagePolicy, error) {
+	logger.Debug("creating package policy...")
+
+	if r.scenario.Version == "" {
+		r.scenario.Version = pkgManifest.Version
+	}
+
+	if r.scenario.Package == "" {
+		r.scenario.Package = pkgManifest.Name
+	}
+
+	pp := kibana.PackagePolicy{
+		Namespace: "ep",
+		PolicyID:  p.ID,
+		Vars:      r.scenario.Vars,
+		Force:     true,
+		Inputs: map[string]kibana.PackagePolicyInput{
+			fmt.Sprintf("%s-%s", r.scenario.DataStream.Name, r.scenario.Input): {
+				Enabled: true,
+				Streams: map[string]kibana.PackagePolicyStream{
+					fmt.Sprintf("%s.%s", pkgManifest.Name, r.scenario.DataStream.Name): {
+						Enabled: true,
+						Vars:    r.scenario.DataStream.Vars,
+					},
+				},
+			},
+		},
+	}
+	pp.Package.Name = pkgManifest.Name
+	pp.Package.Version = r.scenario.Version
+
+	policy, err := r.options.KibanaClient.CreatePackagePolicy(pp)
+	if err != nil {
+		return nil, err
+	}
+
+	return policy, nil
+}
+
+func (r *runner) initializeGenerator() (genlib.Generator, error) {
+	totSizeInBytes, err := humanize.ParseBytes(r.scenario.Corpora.Generator.Size)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := r.getGeneratorConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	fields, err := r.getGeneratorFields()
+	if err != nil {
+		return nil, err
+	}
+
+	tpl, err := r.getGeneratorTemplate()
+	if err != nil {
+		return nil, err
+	}
+
+	var generator genlib.Generator
+	switch r.scenario.Corpora.Generator.Template.Type {
+	default:
+		logger.Debugf("unknown generator template type %q, defaulting to \"placeholder\"", r.scenario.Corpora.Generator.Template.Type)
+		fallthrough
+	case "", "placeholder":
+		generator, err = genlib.NewGeneratorWithCustomTemplate(tpl, *config, fields, totSizeInBytes)
+	case "gotext":
+		generator, err = genlib.NewGeneratorWithTextTemplate(tpl, *config, fields, totSizeInBytes)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return generator, nil
+}
+
+func (r *runner) getGeneratorConfig() (*config.Config, error) {
+	var (
+		data []byte
+		err  error
+	)
+
+	if r.scenario.Corpora.Generator.Config.Path != "" {
```

Comment:
nit: here if we have a both a raw config and a config path, the first will override the second.
maybe move to a switch? one of the two will still have priority, or let's validate that only one of the two is set